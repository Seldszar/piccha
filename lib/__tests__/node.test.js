const Node = require("../node");
const Tree = require("../tree");

describe("Node", () => {
  let tree;

  beforeEach(() => {
    tree = new Tree({
      "/lorem": "ipsum",
      "/lorem/ipsum": "dolor",
    });
  });

  test("should return an existent node", () => {
    expect(new Node(tree, "/lorem")).toMatchSnapshot();
  });

  test("should return an non-existent node", () => {
    expect(new Node(tree, "/ipsum")).toMatchSnapshot();
  });

  test("should return the name of the root node", () => {
    expect(new Node(tree, "/").name).toMatchSnapshot();
  });

  test("should return the name of a child node", () => {
    expect(new Node(tree, "/lorem").name).toMatchSnapshot();
  });

  test("should return the parent of the root node", () => {
    expect(new Node(tree, "/").parent).toMatchSnapshot();
  });

  test("should return the node parent", () => {
    expect(new Node(tree, "/lorem/ipsum").parent).toMatchSnapshot();
  });

  test("should return the child nodes", () => {
    expect(new Node(tree, "/lorem").children).toMatchSnapshot();
  });

  test("should return a child node", () => {
    expect(new Node(tree, "/lorem").get("ipsum")).toMatchSnapshot();
  });

  test("should update the value of the node", () => {
    const node = new Node(tree, "/lorem");

    node.value = "dolor";

    expect(node).toMatchSnapshot();
  });

  test("should update a child node", () => {
    expect(new Node(tree, "/lorem").set("ipsum", "dolor")).toMatchSnapshot();
  });

  test("should delete a child node", () => {
    expect(new Node(tree, "/lorem").delete("ipsum")).toMatchSnapshot();
  });

  test("should return the string representation of the node", () => {
    expect(new Node(tree, "/").toString()).toMatchSnapshot();
  });
});
