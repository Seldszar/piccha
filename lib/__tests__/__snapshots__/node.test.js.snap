// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Node should delete a child node 1`] = `true`;

exports[`Node should return a child node 1`] = `
Object {
  "children": Map {},
  "path": "/lorem/ipsum",
  "value": "dolor",
}
`;

exports[`Node should return an existent node 1`] = `
Object {
  "children": Map {
    "ipsum" => Object {
      "children": Map {},
      "path": "/lorem/ipsum",
      "value": "dolor",
    },
  },
  "path": "/lorem",
  "value": "ipsum",
}
`;

exports[`Node should return an non-existent node 1`] = `
Object {
  "children": Map {},
  "path": "/ipsum",
  "value": undefined,
}
`;

exports[`Node should return the child nodes 1`] = `
Map {
  "ipsum" => Object {
    "children": Map {},
    "path": "/lorem/ipsum",
    "value": "dolor",
  },
}
`;

exports[`Node should return the name of a child node 1`] = `"lorem"`;

exports[`Node should return the name of the root node 1`] = `Symbol(root)`;

exports[`Node should return the node parent 1`] = `
Object {
  "children": Map {
    "ipsum" => Object {
      "children": Map {},
      "path": "/lorem/ipsum",
      "value": "dolor",
    },
  },
  "path": "/lorem",
  "value": "ipsum",
}
`;

exports[`Node should return the parent of the root node 1`] = `null`;

exports[`Node should return the string representation of the node 1`] = `"[object Node]"`;

exports[`Node should update a child node 1`] = `
Object {
  "children": Map {
    "ipsum" => Object {
      "children": Map {},
      "path": "/lorem/ipsum",
      "value": "dolor",
    },
  },
  "path": "/lorem",
  "value": "ipsum",
}
`;

exports[`Node should update the value of the node 1`] = `
Object {
  "children": Map {
    "ipsum" => Object {
      "children": Map {},
      "path": "/lorem/ipsum",
      "value": "dolor",
    },
  },
  "path": "/lorem",
  "value": "dolor",
}
`;
